### DISCUSSION TITLE
Cannot apply fix for single rule

### DISCUSSION THREAD

**Discussion by ralfwelt** [JosefPihrt/Roslynator#1042] [Q&A] [answered]
Hi there and many thanks for this wonderful tool!
I try to apply a fix for a single rule on all projects of a solution. 
(I tried to create an issue with the text below but apparently don't have permission for that)
To achieve this, I created a ruleset with a single rule (linked below)
Unfortunately the file is not taken into account. Instead all available fixes are applied.
Any help appreciated!

**Product and Version Used**: 
roslynator.commandline.0.5.0.nupkg
"Roslynator.Analyzers" Version="4.2.0"

**Steps to Reproduce**:
[roslynator.ruleset.txt](https://github.com/JosefPihrt/Roslynator/files/10795290/roslynator.ruleset.txt)
Use CodeAnalysis ruleset with single rule (as attached)
Fix single project with following command line (works for all projects the same):
C:\_\Build\Roslynator\Roslynator.exe fix c:\_\oldar2\plugins\rawdatareaders\archerreadertranslator\chemstationtoarcher\chemstationtoarcher.csproj -p CodeAnalysisRuleSet=C:\_\Build\PowerShell\DevScripts\Configfiles\roslynator.ruleset.txt -v diag

**Actual Behavior**:
```csharp
[log.txt](https://github.com/JosefPihrt/Roslynator/files/10795334/log.txt)
As seen in the log, roslynator tries to apply all available fixes
```

**Expected Behavior**:
```csharp
```
Only the single fix for RSC1005 should be applied

**Comment by josefpihrt** [comment, marked as answer]
Hi,

if you want to fix only specified set of rules you can use parameter `--supported-diagnostics <DIAGNOSTIC_IDS>`

Your command could look like:

```
C:_\Build\Roslynator\Roslynator.exe fix c:_\oldar2\plugins\rawdatareaders\archerreadertranslator\chemstationtoarcher\chemstationtoarcher.csproj --supported-diagnostics RCS1005 -v diag
```

**Reply by ralfwelt** [reply]
Works like a charm (even less effort than my original solution)
Thank you very much!

**Comment by ralfwelt** [comment]
Great news, thanks Josef!

Will try ...

Regards,
Ralf
